lewis/adapters/stream.py

```diff
@@ -48,20 +48,11 @@ def found_terminator(self):
         self.buffer = []
 
         try:
-            match = None
-            for cmd in self.target.commands:
-                match = cmd.pattern.match(request)
-                if match:
-                    groups = match.groups()
-                    func = getattr(self.target, cmd.method)
-
-                    args = cmd.map_arguments(groups)
-                    reply = cmd.return_mapping(func(*args))
-                    break
-
-            if match is None:
+            try:
```

Comment:
The only thing left here that I do not like is the nested `try...except`. Otherwise `next` seems like a good solution to this problem. I don't really care that we perform the "winning match" twice, it's a small cost to pay for avoiding all the other stuff.

`process_request` still raises an exception now, but a standard one, so we do not need a new type here.